### ISSUE TITLE
question pull request

### ISSUE THREAD

**Issue by kashimAstro** [bakercp/ofxDlib#19]
Hi @bakercp 

I hope all is well.
I'm adding some examples to my fork, can it make sense to do a pull request?  

for the moment only:
semantic segmentation and find cars.

https://github.com/kashimAstro/ofxDlib/tree/master/example_dlib_dnn_semantic_segmentation
https://github.com/kashimAstro/ofxDlib/tree/master/example_dlib_dnn_cars
https://github.com/kashimAstro/ofxDlib/tree/master/example_dlib_dnn_cars2

I'm working on an example for android and 
the transport of an example for speech recognition,
but for the latter it will take a bit.

Ciao!

**Comment by kashimAstro** [comment]
![istantanea_2018-03-28_22-13-50](https://user-images.githubusercontent.com/6746680/38056085-b455a2ce-32db-11e8-93d9-387c11a983f4.png)
![istantanea_2018-03-28_22-41-40](https://user-images.githubusercontent.com/6746680/38056087-b473d37a-32db-11e8-92c6-9b0582331bdd.png)
![istantanea_2018-03-28_23-25-28](https://user-images.githubusercontent.com/6746680/38057346-667ecac2-32df-11e8-9204-c82c5e1fb8ff.png)

**Comment by bakercp** [comment]
Yeah!  Absolutely pull request.  For organizational purposes, I'm trying to keep anything prefixed with `example_dlib_*` to be pretty much a direct translation of the original dlib examples.  I don't yet have a great system for more oF friendly examples (i.e. with better reusable packaging for nets etc like you're making and like I have for [LeNet](https://github.com/bakercp/ofxDlib/blob/master/libs/ofxDlib/include/ofx/Dlib/Network/LeNet.h)).  I'd say propose a naming scheme and just make sure there is a good README with a format similar to what we've been trying to add to all oF examples (e.g. a screen shot, summary, expected output, etc).  Thanks for the work on this!

**Comment by kashimAstro** [comment]
I have more or less respected the prefix of the examples, for the creation of a custom network, I would like to have a kind of struct to generate all the layers dynamically, but I do not think this is simple.

now I take a look at LeNet.

**Comment by kashimAstro** [comment]
yesterday night i pushed a pull request:
https://github.com/bakercp/ofxDlib/pull/20

but maybe I did something wrong, (sorry, i'm not very good with github) what am I doing wrong?
this is the result:

![git](https://user-images.githubusercontent.com/6746680/38084317-b00ed124-334c-11e8-89d8-15d0422374ac.png)

**Comment by kashimAstro** [comment]
Hi @bakercp 

I built an example for android, but this implies a variation of the addon_config.mk,
for the moment i pushed only the example for android-studio in my fork:

https://github.com/kashimAstro/ofxDlib/tree/master/example_android

I loaded on the gist a model of addon_config.mk, (try to take a look):
https://gist.github.com/kashimAstro/d10219bbe0b8667a8426880ad3180215

the precompiled libraries are also needed, i can also load those if you agree.

PS:
for some strange reason android-studio does not allow 
me to compile examples with too many underscores,
for this reason the example is calling only example_android.

**Comment by kashimAstro** [comment]
I am also preparing an example emscripten

**Comment by kashimAstro** [comment]
ok, now we also have a working emscripten example:
https://github.com/kashimAstro/ofxDlib/tree/master/example_emscripten_dlib_face_detector

I have also modified addons_config.mk
adding emscripten and setting the path for static libraries:
https://gist.github.com/kashimAstro/d10219bbe0b8667a8426880ad3180215

I'd like to load static libraries for android* and emscripten

what do you think about it?
I can load on my fork in: 

> libs/dlib/lib/
> <ul>
> <li>libs/dlib/lib/android/arm64-v8a</li>
> <li>libs/dlib/lib/android/armeabi</li>
> <li>libs/dlib/lib/android/armeabi-v7a</li>
> <li>libs/dlib/lib/android/mips</li>
> <li>libs/dlib/lib/android/mips64</li>
> <li>libs/dlib/lib/android/x86</li>
> <li>libs/dlib/lib/android/x86_64</li>
> </ul>
> 
> and 
> <ul>
> <li>libs/dlib/lib/emscripten</li>
> </ul>

and at this point i push addons_config.mk in my fork and then make a pull request..

emscripten:
![webdlib](https://user-images.githubusercontent.com/6746680/38873789-e3c8ac7e-4256-11e8-80bd-feab87ce8fa1.png)

android:
![androiddlib](https://user-images.githubusercontent.com/6746680/38873791-e3e2391e-4256-11e8-9b32-13f8ddb91c88.jpg)

Good day!

**Comment by bakercp** [comment]
Cool!

**Comment by kashimAstro** [comment]
if you want to do a test with the precompiled libraries, you can download here:
http://kashimastro.ns0.it/ofxDlib_lib/ofxdlib_lib.zip

obviously you need the new addon_config.mk:
https://gist.github.com/kashimAstro/d10219bbe0b8667a8426880ad3180215

the linux64 folder contains a static CPU-only library

Ciao
